store/storetest/user_store.go

```diff
@@ -1671,33 +1671,33 @@ func testUserStoreGetByUsername(t *testing.T, ss store.Store) {
 	defer func() { require.Nil(t, ss.Bot().PermanentDelete(u3.Id)) }()
 
 	t.Run("get u1 by username", func(t *testing.T) {
-		result := <-ss.User().GetByUsername(u1.Username)
-		require.Nil(t, result.Err)
-		assert.Equal(t, u1, result.Data.(*model.User))
+		result, err := ss.User().GetByUsername(u1.Username)
+		require.Nil(t, err)
+		assert.Equal(t, u1, result)
 	})
 
 	t.Run("get u2 by username", func(t *testing.T) {
-		result := <-ss.User().GetByUsername(u2.Username)
-		require.Nil(t, result.Err)
-		assert.Equal(t, u2, result.Data.(*model.User))
+		result, err := ss.User().GetByUsername(u2.Username)
+		require.Nil(t, err)
+		assert.Equal(t, u2, result)
 	})
 
 	t.Run("get u3 by username", func(t *testing.T) {
-		result := <-ss.User().GetByUsername(u3.Username)
-		require.Nil(t, result.Err)
-		assert.Equal(t, u3, result.Data.(*model.User))
+		result, err := ss.User().GetByUsername(u3.Username)
+		require.Nil(t, err)
+		assert.Equal(t, u3, result)
 	})
 
 	t.Run("get by empty username", func(t *testing.T) {
-		result := <-ss.User().GetByUsername("")
-		require.NotNil(t, result.Err)
-		require.Equal(t, result.Err.Id, "store.sql_user.get_by_username.app_error")
+		result, err := ss.User().GetByUsername("")
+		require.NotNil(t, err)
+		require.Equal(t, result.Id, "store.sql_user.get_by_username.app_error")
```

Comment:
Here is expected to check the `err` not the `result` you are having a null pointer there.

Reply:
You are doing it all the way around. You should check the `err` not the `result`

Reply:
My apologies